frontend/src/journal/hooks/useJournal.ts

```diff
@@ -54,20 +54,30 @@ export function useUpdateJournal() {
   const [content, setContent] = useState<string>("");
   const [isSaving, setIsSaving] = useState<boolean>(false);
 
+  const [initialLoaded, setInitialLoaded] = useState<boolean>(false);
+
   const [debounceTimer, setDebounceTimer] = useState<NodeJS.Timeout | null>(
     null
   );
 
   useEffect(() => {
-    if (data?.getJournal) {
+    if (id && data?.getJournal) {
       setTitle(data.getJournal.title);
       setContent(data.getJournal.content ?? "");
+      setInitialLoaded(true);
+    } else if (!id) {
+      setTitle("Untitled");
+      setContent("");
+      setInitialLoaded(true);
     }
-  }, [data]);
+  }, [id, data]);
 
   const saveJournal = useCallback(
     async (newTitle: string, newContent: string): Promise<void> => {
-      if (!id) return;
+      if (!id || !initialLoaded) return;
+
+      if (newTitle === title && newContent === content) return;
```

Comment:
Nitpick (explanation: "nitpick" usually means that it's a small style suggestion that is not mandatory)

It's a good practice to always wrap `if` statement with curly braces `{}`

So here it would be 
```
if (newTitle === title && newContent === content) {
  return;
}
```

Having them inline breaks the code reading flow a little bit

Also curious, are you using prettier formatter? If not — use it, it's really cool! You can set rules and when you save the fie, the code will be auto formatted to the style guidelines you set for it

Reply:
Done!